### DISCUSSION TITLE
Choose how to parse a file manually

### DISCUSSION THREAD

**Discussion by zhaob1n** [helix-editor/helix#3261] [Q&A] [answered]
Normally helix parse a file based on the file extensions, like .cpp. Can we specify the way helix treat a file on our own? When I am viewing the source code of STL in C++, the file has no extension thus making the syntax highlight fail. Can I make helix treat it as a cpp file?

**Comment by the-mikedavis** [comment, marked as answer]
You can use the `:set-language` (`:lang` for short) command to tell helix which language you'd like to use for syntax highlight and LSP on a buffer.
